### DISCUSSION TITLE
UI-Testing: How does one perform a button click programmatically?

### DISCUSSION THREAD

**Discussion by berndmehnert** [AvaloniaUI/Avalonia#7383] [Q&A] [answered]
Hi there ;),

what is the correct way to programmatically click on a button in Avalonia? More precisely: 
Let's say we want to write a UI-test and at some point we have to click on a button, which is f.e. given as

`<Button Name="MyButton"   Content="{Binding MyButtonBinding}" Command="{Binding MyFunction}"></Button>`

The thing that should do the job here, we would think, would simply be to fire a click event. I.e. within our test we would write something like

```
Button myButton = myView.FindControl<Button>("MyButton");

var clickEvent = new RoutedEventArgs(Button.ClickEvent);

myButton.RaiseEvent(clickEvent);
```

On the one hand we would easily check that indeed that fires a click event, but on the other hand (at least in my case using Avalonia 0.10.06, Linux) we would see that this is not the same as a "real button click". In my case f.e. the click event had no effect, the click should have triggered "MyFunction" to be called but nothing happened. So what is the correct way to do this? Thank you very much for any answer ...

**Comment by maxkatz6** [comment, marked as answer]
It is the correct way to do it - raise an event. You can raise click event, or keydown, or pointer pressed, or any other that can work there.

**Comment by berndmehnert** [comment]
The keydown event did it ;) Thank you very much ...

**Reply by tyHdev** [reply]
Hello, could you please share the code snippet you used? I am having some difficulty with this same problem. Thanks!

edit: I am using ReactiveUI so for now I realized I can just invoke the bound-command directly on the control
